ros-vm-windows/azuredeploy.json

```diff
@@ -0,0 +1,280 @@
+{
+    "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "location": {
+            "type": "string",
+            "defaultValue": "[resourceGroup().location]",
+            "metadata": {
+              "description": "Location for all resources."
+            }
+        },
+        "projectName": {
+            "type": "string",
+            "maxLength": 8,
+            "metadata": {
+              "description": "Specifies a name for generating resource names."
+            }
+        },
+        "virtualMachineSize": {
+            "type": "string",
+            "defaultValue": "Standard_NV6",
+            "allowedValues": [
+                "Standard_D2",
+                "Standard_NV6",
+                "Standard_NV12",
+                "Standard_NV24"
+            ],
+            "metadata": {
+                "description": "The virtual machine size."
+            }
+        },
+        "adminUsername": {
+            "type": "string",
+            "metadata": {
+                "description": "User name for the Virtual Machine."
+            }
+        },
+        "adminPassword": {
+            "type": "securestring",
+            "metadata": {
+                "description": "Password for the Virtual Machine."
+            }
+        }
+    },
+    "variables": {
+        "location": "[parameters('location')]",
+        "networkInterfaceName": "[concat(parameters('projectName'), '-nic')]",
+        "networkSecurityGroupName": "[concat(parameters('projectName'), '-nsg')]",
+        "virtualNetworkName": "[concat(parameters('projectName'), '-vnet')]",
+        "publicIpAddressName": "[concat(parameters('projectName'), '-ip')]",
+        "virtualMachineName": "[concat(parameters('projectName'), '-vm')]",
+        "diagnosticsStorageAccountName": "[concat(uniqueString(resourceGroup().id),'diag')]",
+        "nsgId": "[resourceId(resourceGroup().name, 'Microsoft.Network/networkSecurityGroups', variables('networkSecurityGroupName'))]",
+        "vnetId": "[resourceId(resourceGroup().name,'Microsoft.Network/virtualNetworks', variables('virtualNetworkName'))]",
+        "subnetName": "default",
+        "subnetRef": "[concat(variables('vnetId'), '/subnets/', variables('subnetName'))]",

```

Comment:
remove concat and just use resourceId

[resourceId(resourceGroup().name,'Microsoft.Network/virtualNetworks/subnets', variables('virtualNetworkName'), variables('subnetName'))]